src/Umbraco.Infrastructure/Examine/ExamineIndexRebuilder.cs

```diff
@@ -49,180 +52,149 @@ public bool CanRebuild(string indexName)
         return _populators.Any(x => x.IsRegistered(index));
     }
 
+    /// <inheritdoc/>
+    [Obsolete("Use RebuildIndexAsync() instead. Scheduled for removal in v18.")]
     public virtual void RebuildIndex(string indexName, TimeSpan? delay = null, bool useBackgroundThread = true)
+        => RebuildIndexAsync(indexName, delay, useBackgroundThread).GetAwaiter().GetResult();
+
+    /// <inheritdoc/>
+    public virtual async Task<Attempt<IndexRebuildResult>> RebuildIndexAsync(string indexName, TimeSpan? delay = null, bool useBackgroundThread = true)
     {
-        if (delay == null)
-        {
-            delay = TimeSpan.Zero;
-        }
+        delay ??= TimeSpan.Zero;
 
         if (!CanRun())
         {
-            return;
+            return Attempt.Fail(IndexRebuildResult.NotAllowedToRun);
         }
 
-        if (useBackgroundThread)
-        {
-            _logger.LogInformation("Starting async background thread for rebuilding index {indexName}.", indexName);
-
-            _backgroundTaskQueue.QueueBackgroundWorkItem(
-                cancellationToken =>
-                {
-                    // Do not flow AsyncLocal to the child thread
-                    using (ExecutionContext.SuppressFlow())
-                    {
-                        Task.Run(() => RebuildIndex(indexName, delay.Value, cancellationToken));
+        Attempt<Guid, LongRunningOperationEnqueueStatus> attempt = await _longRunningOperationService.RunAsync(
+            GetRebuildOperationTypeName(indexName),
+            async ct =>
+            {
+                await RebuildIndex(indexName, delay.Value, ct);
+                return Task.CompletedTask;
+            },
+            allowConcurrentExecution: false,
+            runInBackground: useBackgroundThread);
 
-                        // immediately return so the queue isn't waiting.
-                        return Task.CompletedTask;
-                    }
-                });
-        }
-        else
+        if (attempt.Success)
         {
-            RebuildIndex(indexName, delay.Value, CancellationToken.None);
+            return Attempt.Succeed(IndexRebuildResult.Success);
         }
+
+        return attempt.Status switch
+        {
+            LongRunningOperationEnqueueStatus.AlreadyRunning => Attempt.Fail(IndexRebuildResult.AlreadyRebuilding),
+            _ => Attempt.Fail(IndexRebuildResult.Unknown),
+        };
     }
 
+    /// <inheritdoc/>
+    [Obsolete("Use RebuildIndexesAsync() instead. Scheduled for removal in v18.")]
```

Comment:
I guess this should be indicated for removal in 19 now (if this PR is indeed targeted for 17).